### DISCUSSION TITLE
Understanding "Project"

### DISCUSSION THREAD

**Discussion by buhtz** [geany/geany#3241] [General]
I am using Geany with Python files.

Today I tried the projects feature the first time. And I assume I misunderstand its intention.

Right clicking into the code on a method or function I can find it's declaration in another file **when this file is open** in another tap.

But when the file with the declaration is not open the declaration is not found. That file also dissapear from the side panels project tree.

I would have expected that the "project" does remember which files belong to it. And when I search for a declaration the search is done over all files from the project **no matter if they are currently open or not**.

Is this possible?

**Comment by elextr** [comment]
The Geany "project" concept is more lightweight than that of full fat IDEs like Visual Studio or Eclipse.  It is basically a way of remembering multiple different file sets that the user is working on and allowing switching between them.  So closing a file removes it from the "project".

The various project plugins provide additional functionality that is closer to full fat IDEs, Project Organiser for example assumes all files in the project directory are in the project and loads symbols for them even if they are closed, so inter-file declarations should be found.  Probably the first thing you should try.

**Comment by buhtz** [comment]
Thanks for explanation.
I don't have admin rights on my system. Is there a way to "install" the plugnis as a user?

The tar-balls need to be build. Am also not able to do that.
The exe files are archives. But I don't know which one of its content should I copy somewhere.

Btw: Notepad++ is able to install plugins without admin rights.

**Comment by elextr** [comment]
You havn't said what OS you are using, but I take it from the mention of Notepad++ its Windows.  IIUC you can set where to install Geany using the `setup.exe` installer, so you can re-install Geany in your own directory, then install geany-plugins to the same place. 

But AFAIK there is no guarantee that there are not libraries that need to go to system directories, so if the above doesn't work there is a possible option B. IF (big IF, I don't know Windows requirements at all) the plugin needs no libraries other than what Geany uses then you might be able to just copy the dll from the `setup.exe` using an archive manager.  Even on Linux I can read the contents of the `setup.exe` and navigate to the `lib/geany/projectorganiser.dll`.  As best I can tell its the only thing projectorganiser needs, but the installer has a lot of system libraries that it might need.

There is a setting in `Edit->Preferences->General->Startup->Extra Plugin Path` that you can set in the system installed Geany to point to a non-system directory you can write the `.dll` file to.  Hopefully it will work, but no guarantees on Windows.
